### PR TITLE
Project Retrieval endpoints return Likes

### LINKED ISSUE #329: Return project likes on the getProject endpoints
**Is your feature request related to a problem? Please describe.**
There is no way to get the amount of likes a project has, and check if a user has liked a project.

**Describe the solution you'd like**
I think it would be good if we include both of these in the projectResultResource for the getAllProjects and getProject endpoints.

This issue is related to this closed issue: #229

### PR DESCRIPTION
**Pull request by Brend-Smits**
## Description

Likes were not being returned when retrieving projects. Due to performance reasons, we are just returning an array for now with user id + date instead of Like Count and boolean to check if the currently logged-in user has liked that specific project. 
In the future, we should probably move the like and follow things into their own controller with respective services. The frontend can then make a call to retrieve project metadata, which can take a little longer and should be non-blocking for the frontend. 
For now, the frontend can count the objects in the likes array and check if a user-id is equal to the one that is logged in.
Fixes #329.

#335 should be merged before this as I merged that branch in this one to investigate performance while retrieving likes.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Make a call to project overview.
2. Check if object returns likes (make sure you add likes into DB)

## Link to issue

Closes: #329

### CODE REVIEW

**Comment by Brend-Smits** [comment]
Integration check is failing based on performance external wizard call, just FYI

**Review by waltersajtos — DISMISSED** [review]

File: Repositories/ProjectRepository.cs [hunk 1]
```diff
@@ -287,44 +228,153 @@ public override void Update(Project entity)
...
+
+            if(project?.User?.IsPublic == false)
+            {
+                project.User.Email = Defaults.Privacy.RedactedEmail;
+            }
+            return project;
+        }
+
+        /// <summary>
+        ///     Redact user email from the User if isPublic setting is set to false
```
Comment (Ruby77) [inline Repositories/ProjectRepository.cs:249]: This is an example you didn't add a dot at the end of the sentence

**Review by Ruby77 — CHANGES_REQUESTED** [review]
The code looks really clean. I'm sorry to nitpick, but in your documentation, you sometimes end the sentence with a dot, and sometimes you don't do this. According to the code guidelines
`End comment text with a period.` I suggest do end it with a dot all the time (if it's a correct sentence. On my first PR I got some great discussion about this, and we decided to only end the sentence with a dot whenever it's a correct sentence.)

File: Repositories/ProjectRepository.cs (hunk 1 above)
Comment (Brend-Smits) [reply Repositories/ProjectRepository.cs]: Thanks, you're right. I kinda forget sometimes, which is why we have these pull requests ^^

**Review by Ruby77 — APPROVED** [review]

**Review by waltersajtos — APPROVED** [review]

**Pull request merged**